crates/router/src/core/admin.rs

```diff
@@ -389,7 +389,22 @@ impl MerchantAccountCreateBridge for api::MerchantAccountCreate {
             .await?;
 
         let merchant_account_type = match organization.get_organization_type() {
-            OrganizationType::Standard => MerchantAccountType::Standard,
+            OrganizationType::Standard => {
+                match self.merchant_account_type {
+                    // Allow only if explicitly Standard or not provided
+                    Some(MerchantAccountRequestType::Standard) | None => {
+                        MerchantAccountType::Standard
+                    }
+                    Some(_) => {
```

Comment:
Please avoid using wildcard(_) in match arms.
```suggestion
                    Some(MerchantAccountRequestType::Connected) => {
```